### DISCUSSION TITLE
[Suggestion]: focus on vlang core instead of peripheral products

### DISCUSSION THREAD

**Discussion by axetroy** [vlang/v#11496] [General]
Currently, The `V` is still under development, and v1.0.0 is still in the foreseeable future.

But at the same time, V official team has developed too many great peripheral products (UI/editor/Unix system/LSP, and so on)

This leads to too much energy being allocated to these projects.

I can understand that a successful product can help build a reputation, like `Docker` for `Golang`.

But is this too early when V is not ready?

There are still many BUGs in the current V core, and many functions of the standard library are missing.

For most people, if you want to develop a tool (library), most have to implement it yourself, which is a stumbling block that hinders the growth of the community. Most people just stop here and choose another language.

Otherwise, for users, it may take 5 years and more for V to release v1.0.0.

This is just a suggestion, thank you for all your great work. ❤️

**Comment by ghost** [comment]
I agree with your opinion, but unfortunately it will be impossible.
As I see it, it is due to the following reasons

- There is no clear language specification to be implemented.
- Each person is freely contributing the code they want, and there is no centralized governing.

It will take more than five years to reach 1.0.0.
However, most of the bugs can be worked around in some way, so you won't have any trouble coding for the time being.
